### PR TITLE
impl(w1r3): deploy Java runners

### PR DESCRIPTION
**Pull request by coryan**

### CODE REVIEW

File: w1r3/infra/mig/java/main.tf [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    Environment="HOME=/home/cloudservice"
+    ExecStartPre=/usr/bin/docker-credential-gcr configure-docker ; /usr/bin/docker pull gcr.io/${var.project}/w1r3/java:latest
+    ExecStart=/usr/bin/docker run --rm -u 2000 --name=w1r3-java-%i gcr.io/${var.project}/w1r3/java:latest java -jar /r/w1r3.jar -project-id ${var.project} -bucket ${var.bucket} -iterations ${local.iterations} -deployment mig -workers 1
+    ExecStop=/usr/bin/docker stop w1r3-java-%i
+    ExecStopPost=/usr/bin/docker rm w1r3-java-%i
+
+runcmd:
+- docker-credential-gcr configure-docker
+- systemctl daemon-reload
+- for i in $(seq 1 ${local.cpus}); do systemctl start w1r3@$${i}.service; done
```
Comment (BenWhitehead) [inline w1r3/infra/mig/java/main.tf:96]: To confirm my understanding of the systemd unit and the cloud config:

The unit will start an individual process with a single worker. Then cloud-config will start multiple services, one for each CPU on the host machine?

File: w1r3/infra/mig/java/main.tf [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  region = var.region
+  version {
+    instance_template = google_compute_instance_template.default.id
+    name              = "primary"
+  }
+  base_instance_name = local.base_instance_name
+  target_size        = 3
+
+  # NOTE: the name of this resource must be unique for every update;
+  #       this is wy we have a app_version in the name; this way
```
Comment (BenWhitehead) [inline w1r3/infra/mig/java/main.tf:134]:
```suggestion
  #       this is why we have a app_version in the name; this way
```

**Review by BenWhitehead — APPROVED** [review]

File: w1r3/infra/mig/java/main.tf (hunk 1 above)
Comment (coryan) [reply w1r3/infra/mig/java/main.tf:96]: Yes.  The idea is that measuring CPU / byte is easier if a single thread is doing the uploads and downloads.

Comment (BenWhitehead) [reply w1r3/infra/mig/java/main.tf:96]: Makes sense, thanks for confirming

**Pull request merged**